src/Build/Definition/Project.cs

```diff
@@ -2479,6 +2479,10 @@ private List<GlobResult> GetAllGlobs(List<ProjectItemElement> projectItemElement
 
                     if (!string.IsNullOrEmpty(itemElement.Include))
                     {
+                        if (NativeMethodsShared.IsWindows && FileUtilities.InvalidFileNameChars.Any(c => itemElement.Include.Contains(c)))
```

Comment:
Can you avoid Linq, since it's perf sensitive here? This is less overhead-
` if (NativeMethodsShared.IsWindows && itemElement.Include.IndexOfAny( FileUtilities.InvalidFileNameChars) != -1`

Reply:
I probably should have sent this as a draft. I'm planning to use it as a starting point for "here's how we might implement this if we want to do this at all" tomorrow.